crates/typst-pdf/src/gradient.rs

```diff
@@ -343,14 +353,14 @@ fn register_gradient(
 ///
 /// Structure:
 ///  - flag: `u8`
-///  - points: `[u16; 24]`
-///  - colors: `[u16; 12]`
+///  - points: `[u16; 32]`
+///  - colors: `[u16; 16]`
```

Comment:
I believe the points should stay the same, as there are 12 `[u16; 2]` arrays, so a total of 24 `u16` values. The data for the four written colors `[c0, c0, c1, c1]` depends on the number of components, as there aren't always four components.

```suggestion
///  - points: `[u16; 24]`
///  - colors: `[u16; 4*N]` (N = number of components)
```

Reply:
Good point! Corrected in the latest commit.